### ISSUE TITLE
New sync default values don't account for plugin deprecations

### ISSUE THREAD

**Issue by scirner22** [Kong/kubernetes-ingress-controller#2367]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I understand https://github.com/Kong/kubernetes-ingress-controller/pull/2286 moved the way default diffs are calculated and that's been a blessing for us during loop iterations when a diff is detected as it would previously update a bunch of resources that didn't need to be.

One default value that I think is still incorrect is with the [acl](https://docs.konghq.com/hub/kong-inc/acl/) and [ip-restriction](https://docs.konghq.com/hub/kong-inc/ip-restriction/) plugins. In kong 2.1 `whitelist/blacklist` were deprecated in favor of `allow/deny`. For this reason, we already moved all of our plugins of those types to the new syntax, unfortunately, this means we get no-op updates quite frequently like below. We have so many of these in our cluster we actually get intermittent increased latency on our ingresses.

```diff
updating plugin ip-restriction for route 6fbd119f-7cf3-4790-a6de-0617b0656df5  {
   "config": {
-    "allow": [
-      "redacted",
-      "redacted",
-      "redacted",
-      "redacted",
-      "redacted",
-      "redacted"
-    ],
+    "allow": null,
     "deny": null,
     "message": null,
     "status": null
+    "whitelist": [
+      "redacted",
+      "redacted",
+      "redacted",
+      "redacted",
+      "redacted",
+      "redacted"
+    ]
   },
   "enabled": true,
   "id": "6a28d2db-cba8-4137-94a2-6d720eda1bf0",
   "name": "ip-restriction",
   "protocols": [
     "http",
     "https"
   ],
   "route": {
     "id": "6fbd119f-7cf3-4790-a6de-0617b0656df5"
   },
   "tags": [
     "managed-by-ingress-controller"
   ]
 }
```

I looked around the kic, deck, and go-kong code but it wasn't immediately apparent where the default schemas are derived from. [Here](https://github.com/Kong/kong/blob/master/kong/plugins/ip-restriction/schema.lua) is the kong lua schema, but since that's the accurate version, I assume that's not what is being used.

### Expected Behavior

No diff detected during the sync loop.

### Steps To Reproduce

```markdown
apiVersion: configuration.konghq.com/v1
config:
  allow:
  - 0.0.0.0
kind: KongPlugin
metadata:
  name: test-ip-restriction
  namespace: testing
plugin: ip-restriction


Apply the above and then perform any action that results in an update (Adding a second resource should trigger it).
```

### Kong Ingress Controller version

```shell
kic: A nightly version that includes up until #2338 (2.3)
kong: 2.7.0
```

### Kubernetes version

```shell
1.20
```

**Comment by rainest** [comment]
Do you see this on the 2.3.0 release from today? It includes the deck default change, but I'm not able to replicate this. With:

```
apiVersion: configuration.konghq.com/v1
config:
  allow:
  - 0.0.0.0/0
kind: KongPlugin
metadata:
  creationTimestamp: "2022-04-05T22:16:33Z"
  generation: 1
  name: kong-id
  namespace: default
  resourceVersion: "11120"
  uid: f1e20e99-bd9d-4243-9fc2-9418d633571c
plugin: ip-restriction
```
I see the plugin created initially, and then on a subsequent change to the Ingress path, only see the Ingress path update, as expected:

```
time="2022-04-05T22:22:24Z" level=debug msg="converting configuration to deck config"
time="2022-04-05T22:22:24Z" level=debug msg="sending configuration to Kong Admin API"
creating plugin ip-restriction for route cb3f1d98-9184-431f-a774-bc5961b6e9d7
time="2022-04-05T22:22:24Z" level=info msg="successfully synced configuration to kong."

time="2022-04-05T22:22:42Z" level=debug msg="converting configuration to deck config"
time="2022-04-05T22:22:42Z" level=debug msg="sending configuration to Kong Admin API"
updating route default.httpbin.00  {
   "https_redirect_status_code": 426,
   "id": "cb3f1d98-9184-431f-a774-bc5961b6e9d7",
   "name": "default.httpbin.00",
   "path_handling": "v0",
   "paths": [
-    "/httpbina"
+    "/httpbin"
   ],
   "preserve_host": true,
   "protocols": [
     "http",
     "https"
   ],
   "regex_priority": 100,
   "request_buffering": true,
   "response_buffering": true,
   "service": {
     "id": "2cfa7734-0b5a-4938-a854-4c33c1df766e"
   },
   "strip_path": false,
   "tags": [
     "managed-by-ingress-controller"
   ]
 }

time="2022-04-05T22:22:42Z" level=info msg="successfully synced configuration to kong."
```
These plugins do have [a soon-to-be-removed `shorthand_fields` in their config schema](https://github.com/Kong/kong/blob/2.8.0/kong/plugins/ip-restriction/schema.lua#L16-L32) that allows using the old names:

```
$ curl -s localhost:8001/schemas/plugins/ip-restriction | jq .fields[1].config.shorthand_fields    
[
  {
    "blacklist": {
      "elements": {
        "is_regex": true,
        "type": "string"
      },
      "type": "array"
    }
  },
  {
    "whitelist": {
      "elements": {
        "is_regex": true,
        "type": "string"
      },
      "type": "array"
    }
  }
]
```
but I'm not seeing any effect from that on the diff. I believe what you're seeing is the result prior to [this change](https://github.com/Kong/go-kong/commit/2f7c7803805bd4fe2c3bc64f0dfca36f088a6de6#diff-af8350ebad15aeeb0a2561433d1d0907140df8e5b8267a0d0f05bb53d0d53fcb): the original default handler used the entire schema for all objects, but this isn't what we want for plugins, and the original iteration would attempt to add fields from the plugin type itself (`protocols`, `elements`) into the config. `shorthand_fields` is sort of one of these. The fixed version uses `config.fields`, so `config.shorthand_fields` will be excluded.

**Comment by scirner22** [comment]
Thanks! That recent change will more than likely fix this. It appears that it was fixed in [v0.29.0](https://github.com/Kong/go-kong/releases/tag/v0.29.0). I'll attempt to upgrade to kic 2.3.0 since it is now released. Right now I built a custom image that appears to be on go-kong v0.28.1 and kic 0ada2af2b64f6a0c82930f554af4811ee20be3dc

**Comment by shaneutt** [comment]
Sounds good thank you @scirner22. For the moment we consider this resolved, but if you find that with the latest release you still have issues let us know and we'll re-open. :+1:
